src/main/java/xyz/duncanruns/julti/Julti.java

```diff
@@ -122,6 +122,22 @@ public static boolean isRanFromAlternateLocation() {
         return !Paths.get("").toAbsolutePath().equals(getSourcePath().getParent());
     }
 
+    // Uses portion https://stackoverflow.com/questions/35129457/how-to-check-if-a-process-is-running-on-windows
+    // CMD help from Jojoe, thank you jojoe
+    public static void isWallAHKOpen() throws IOException {
+        String tasksCmd = System.getenv("windir") + "/system32/wbem/wmic.exe path Win32_Process where " +
+                "\"CommandLine Like '%AutoHotkey.exe% %TheWall.ahk%'\" get ProcessId /format:list";
+        Process p = Runtime.getRuntime().exec(tasksCmd);
+        BufferedReader input = new BufferedReader(new InputStreamReader(p.getInputStream()));
+
+        ArrayList<String> pids = new ArrayList<String>();
+        String pid = null;
+        while ((pid = input.readLine()) != null)
+            pids.add(pid);
+        pids.removeAll(Arrays.asList("", null));
+        if (pids.size() > 1)
+            log(Level.WARN, "The Macro for Specnr's Wall is open. To prevent issues, close the ahk script.");
```

Comment:
Usage of list over-complicates things.
Instead of building a list and checking if any items exist at the end, you can stop checking when a valid item is added.
```java
String pid = null;
while ((pid = input.readLine()) != null){
    if(pid.isEmpty()) continue; // pid cannot be null as that is checked in the while loop condition
    log(Level.WARN, "The Macro for Specnr's Wall is open. To prevent issues, close the ahk script.");
    input.close(); // not sure if this is necessary
    return;
}
```

Reply:
I noticed that my suggested change would actually break it because I failed to interpret the intentions of `pids.size() > 1`, I have updated my version to account for this while still not having to use a list:

```java
        String pid;
        boolean firstFound = false;
        while ((pid = input.readLine()) != null) {
            pid = pid.trim();
            if (pid.isEmpty()) {
                continue; // pid cannot be null as that is checked in the while loop condition
            }
            if (!firstFound) {
                firstFound = true;
                continue;
            }
            Julti.log(Level.WARN, "MultiResetWall is open. To prevent issues, please close the ahk script.");
            input.close(); // not sure if this is necessary
            return;
        }
```